src/Service/ApiClient.php

```diff
@@ -0,0 +1,50 @@
+<?php
+declare (strict_types=1);
+
+namespace Service;
+
+
+class ApiClient {
+
+    /** @var \Maleficarum\Client\Http\Rest\BasicClient  */
+    protected $httpClient;
+
+    /**
+     * ApiClient constructor.
+     *
+     * @param \Maleficarum\Client\Http\Rest\BasicClient $httpClient
+     */
+    public function __construct(\Maleficarum\Client\Http\Rest\BasicClient $httpClient) {
+        $this->httpClient = $httpClient;
+    }
+
+    /**
+     * @param int $articleId
+     *
+     * @return array
+     */
+    public function getArticle(int $articleId): array {
+        $this->httpClient->get('/articles/'.$articleId);
+
+        return $this->httpClient->getParsedBody();
+    }
+
+    /**
+     * @param int $limit
+     * @param int $offset
+     *
+     * @return array
+     */
+    public function getArticles($limit = 10, $offset = 0): array {
```

Comment:
po pierwsze brak typeHintów.

po drugie z doświadczenia nie używam X argumentów jako typy proste tylko do każdej metody wkładam DTO. W tym wypadku np ArticleFilter (ma to tez zastosowanie dla metody getArticle w której zrobisz setId a metody setLimit i setOffset nie zostaną wywołane). 

Oczywiście ten DTO może się składać z kliku np PaginationFilter, którego będziesz mógł użyć w innych miejscach. Ma to jedną ogromną zalete wszystkie endpointy są spójne by design i nie trzeba tego utrzymywać. Przy wpisywaniu tego z palca raz ktoś użyje limit/offset ktoś inny page a ktoś inny zamieni te parametry miejscami i robi się burdel. Dodatkowo można dodać walidację i ewentualną logikę takiego obiektu (np  jak ktoś przekaże -1 jako limit to dostaje wyniki bez paginacji - spójnie w jednym miejscu a nie ify w każdym endpoincie który może coś takiego obsłużyć).